SAMtemplates/alarms/main.yaml

```diff
@@ -81,3 +85,24 @@ Resources:
         - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
       OKActions:
         - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+
+  PSUConcurrencyAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties: 
+      AlarmName: !Sub "${StackName}-Concurrency-Alarm"
+      AlarmDescription: "Monitors the concurrency of the PSU as a whole and triggers when concurrency exceeds the threshold."
+      Namespace: "AWS/Lambda"
+      MetricName: "ConcurrentExecutions"
+      Statistic: Sum
+      Period: 60 # seconds
+      EvaluationPeriods: 1
+      Threshold: !Ref ConcurrencyThreshold
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: notBreaching
```

Comment:
For this one we might want to treat missing data as an error, its worth checking the metric in cloudwatch and looking if it reports 0 or nothing when the system isnt being used, if the former than we probably want to error if its missing as it suggests some disruption to the monitoring.

Reply:
It does indeed register as 0 when the system is idle. I've made the change - though, we may want to disable the alarm for REF if it gets enabled there!

Reply:
It should be ok, it should only alert when theres a state change. So will probably only error for no data when first deployed.